superset-frontend/src/dashboard/actions/hydrate.js

```diff
@@ -256,6 +258,19 @@ export const hydrateDashboard =
         layout[layoutId].meta.sliceName = slice.slice_name;
       }
     });
+
+    // make sure that parents tree is built
+    if (
+      Object.values(layout).some(
+        element => element.id !== DASHBOARD_ROOT_ID && !element.parents,
+      )
+    ) {
+      updateComponentParentsList({
```

Comment:
Some sample dashboards don't have `parents` in their layout metadata, which caused e2e tests to fail